### ISSUE TITLE
[Community] Developer experience on Kubernetes (as a platform) - Blog

### ISSUE THREAD

**Issue by loujaybee** [cncf/tag-app-delivery#541]
### Contribution Description

I am creating a webinar for work ([source](https://www.gitpod.io/webinars/adopt-kubernetes-for-platform)). The content of the webinar would make for a great CNCF branded blog post. The first draft of that content will be ready next week in preparation for the webinar run date of **15th February 2024**.

The description of the webinar content includes: 
* Build a platform around your Kubernetes cluster(s) that prioritises developer experience
* Avoid “works on my machine” and configuration drift problems with Kubernetes
* Design developer experience for fast feedback (not 30 minute CI builds).
* Go from a single local docker container, to multi-container to pre-prod preview clusters. 
* Use proxies like Telepresence, MirrorD to connect local services to a Kubernetes cluster.
* Considerations on internal developer portals like Backstage. 
...etc

**Previous message in the CNCF Platform Group** - and [link to thread](https://cloud-native.slack.com/archives/C020RHD43BP/p1706222793421969).

> Developer experience on Kubernetes (as a platform) - I’ve been going deep on the topic of end-user developer experience on Kubernetes, including the many, many tools that can form a part, or a whole solution here. A few people expressed they’d like to see a whitepaper on this. Whilst I could take a swing myself, that feels like a topic very much in this groups wheelhouse. If anyone is interested, I’d be keen to explore what this could look like. I see scattered content on the topic, I’m yet to read anything that’s truly comprehensive… 

### Related Working Group (WG)

Platforms

### Contribution type

Blog (Default)

### Why TAG App Delivery?

Very Platform related.

### Related projects/technologies

https://mirrord.dev
https://kubernetes.io
https://www.telepresence.io
https://backstage.io
https://score.dev

### Affiliation disclosure

I work for ([Gitpod](https://gitpod.io/)). If relevant the topic could cover Cloud Development Environments (CDEs), but this content is not intended to centre around CDEs. 

### Additional collaborators

Antoine Bermon, Cat Morris, Abby Bangser, Saim Safdar,

**Comment by loujaybee** [comment]
Scheduled a 30 Minute open invite to start the discussion - [meeting link](https://meet.google.com/pap-xtto-ema ).

**Time:** 4:00pm - 4:30pm, Wednesday, January 31, 2024 - UK, Ireland, Lisbon Time

**Comment by abangser** [comment]
I think this is great @loujaybee !

It sounds like you have a lot of momentum behind this for a webinar and that you could repurpose for a blog post. In general community blog posts are encouraged to be author led so long as they abide by the [community guidelines](https://tag-app-delivery.cncf.io/contribute/community-post-guidelines/). But of course authors can choose to collaborate as much as they would like with the group and that would be great.

One thing I would recommend for this post in particular is to keep an eye on the webinar and determine if there is a natural split for multiple blog posts. I suspect there may be a couple here and that they can help you (and the group) suss out if there is a larger scale white paper or reference architecture here that can grow from them.

**Comment by ebcarty** [comment]
@abangser are there any existing reference architecture(s)?  I believe [cnoe-interest](https://cloud-native.slack.com/archives/C05TN9WFN5S) is working on one (but perhaps focused on AWS).

**Comment by Rotfuks** [comment]
Really interesting topic and I'd like to join in on the discussion. 

Last year I already wrote a blog post on how to Maximize Value with Kubernetes-as-a-Product by shifting the focus to the enduser. It's way less technical and looks more on the exploration and product side of things but I guess that might be a helpful perspective here as well.   
But I also recently did a Telepresence Introduction 1pager for a german print magazine, so I also have some opinions there :laughing:  
What do you think, @loujaybee ?

**Comment by abangser** [comment]
@ebcarty great question!

There are definitely reference architectures including [CNOE](https://cnoe.io/), [BACK](https://github.com/crossplane-contrib/back-stack), [KubeFirst](https://kubefirst.io/), and I am sure many more. Though I am not sure that is the angle @loujaybee is going with this as those are more about the low level tools. That being said, I am looking forward to hearing more and very well could be good to explore those directions!

**Comment by loujaybee** [comment]
**Goal:** - We are driving towards a first blog post to publish for the CNCF covering this topic. 

**Update:** We had an introductory call in the working group where we spoke about Kubernetes and Developer Experience in a very broad and open sense. As there are lots of topics that come under "developer experience and kubernetes". I spoke with @abangser and we agree'd to try and tighten the scope of the discussions to make it easier to collaborate. 

**Scope:** - This blog post: **[config drift causes a painful death for developers](https://score.dev/blog/warning-config-drift-causes-a-painful-death-for-developers)** is a great framing of the problem space. 

**Strawman:** I'm attaching here a strawman that lays out some different "patterns" for working with Kubernetes and resolving challenges with developer experience. Publishing a refined variation of this would be a good first goal here. 

<img width="967" alt="image" src="https://github.com/cncf/tag-app-delivery/assets/5528307/a3656ab9-020e-47cf-a8c6-f2e2d01dd51a">

**Next steps:** - I’ll schedule another call in a week, or so. I'm presenting the content [in a webinar](https://www.gitpod.io/webinars/adopt-kubernetes-for-platform) on Wednesday (join if you'd like). I’ll continue my own research until then. If anyone wants to chat 1-1 this or next week, [here’s my calendly](https://calendly.com/lou-gitpod/chat).

Thanks for all the inputs so far ! Let's continue the discussion here async if needed, too! 👇

**Comment by loujaybee** [comment]
> @abangser One thing I would recommend for this post in particular is to keep an eye on the webinar and determine if there is a natural split for multiple blog posts. I suspect there may be a couple here and that they can help you (and the group) suss out if there is a larger scale white paper or reference architecture here that can grow from them.

Yep! I'm publishing two at work, one articulates this issue of "configuration drift" (Similar to the blog post above ["Warning: config drift causes a painful death for developers"](https://score.dev/blog/warning-config-drift-causes-a-painful-death-for-developers)). And another that talks about patterns [this blog post](https://codefresh.io/blog/okteto/) covers similar content.

**Comment by loujaybee** [comment]
> @Rotfuks Last year I already wrote a blog post on how to Maximize Value with Kubernetes-as-a-Product by shifting the focus to the enduser. It's way less technical and looks more on the exploration and product side of things but I guess that might be a helpful perspective here as well. 

For sure, please share here, those seem very related !

**Comment by Rotfuks** [comment]
Heya @loujaybee! Finally managed to rework the blog post, here's the new version I talked about: https://www.giantswarm.io/blog/maximizing-value-with-kubernetes-as-a-product-fulfilling-the-promise-of-the-cloud
